lib/tooling/base-tool.js

```diff
@@ -45,6 +45,12 @@ export class BaseTool {
         return this.tool.type || 'independent';
     }
 
+    getUniqueFilePrefix() {
+        const timestamp = process.hrtime();
+        const timestamp_str = '_' + timestamp[0] * 1000000 + timestamp[1] / 1000;
+        return this.tool.id + timestamp_str + "_";
```

Comment:
I can't think of any current cases where `tool.id` would not be filesystem safe, but I also don't know if this has been explicitly stated to be fact in the past.

I don't have an issue with requiring `id` be filesystem safe - we just need to make sure that is understood.

Reply:
I added a regex to knock off anything except alphanumeric